Angular/src/app/admin-table/admin-table.component.ts

```diff
@@ -18,75 +17,112 @@ export class AdminTableComponent implements OnInit {
       }
     }
-    return {"Role":"+", "ID":-1, "BandId":band.band_id, "CapabilityId": cap.capability_id}
+
+    return { "Role": "Add new role", "ID": -1, "BandId": band.band_id, "CapabilityId": cap.capability_id }
   }
 
   async switchModal(selectedRole) {
-    if(selectedRole.ID == -1){
-      this.modalService.openAddRoleModal(selectedRole.BandId, selectedRole.CapabilityId, this.departmentId).then(data => {
-        this.displayAlert(data)
-      })
-    }
-    else {
-      this.modalService.openRoleInfoModal(selectedRole.ID);
+    if (selectedRole.ID == -1) {
+      this.openAddRoleModal(selectedRole.BandId, selectedRole.CapabilityId);
+    } else {
+      this.openEditRoleModal(selectedRole.ID);
     }
   }
 
-  openCapabilityModal(){
-    this.modalService.openAddCapabilityModal(this.departmentId).then(data => {
-      this.displayAlert(data)
+  async openRoleInfoModal(selectedRoleId) {
+    await this.dataService.getRoleInformation(selectedRoleId).then(response => {
+      const modalRef = this.modalService.open(RoleInformationComponent);
+      modalRef.componentInstance.roleToDisplay = response[0];
+    });
+  }
+
+  async openBandInfoModal(selectedBandId) {
+    await this.dataService.getBandInformation(selectedBandId).then(response => {
+      const modalRef = this.modalService.open(BandInformationComponent);
+      modalRef.componentInstance.bandToDisplay = response[0];
     });
   }
 
-  loadRoles(){
-    this.dataService.getCapabilityNamesByDepartment(this.departmentId).then(response => {
+  loadRoles() {
+    this.dataService.getCapabilityNamesByDepartment(this.id).then(response => {
       this.capabilities = response;
     });
-    this.dataService.getRolesInDepartment(this.departmentId).then(response => {
+
+    this.dataService.getRolesInDepartment(this.id).then(response => {
       this.jobsInDep = response;
     });
+
     this.dataService.getBands().then(response => {
       this.bands = response;
     });
-    this.dataService.getDepartmentDetails(this.departmentId).then(response => {
+
+    this.dataService.getDepartmentDetails(this.id).then(response => {
       this.departmentName = response;
     });
   }
 
-  displayAlert(data){
-    if(data.hasOwnProperty('success')){
-      this.showAlert = true;
-      this.alertMessage = data.success;
-      this.alertType = "success";
-    } else {
-      this.showAlert = true;
-      this.alertMessage = data.error;
-      this.alertType = "danger";
-    }
-    this.loadRoles();
+  openAddRoleModal(roleBandId, roleCapabilityId) {
+    const modalRef = this.modalService.open(AddRoleModalComponent);
+    modalRef.componentInstance.departmentId = this.id;
+    modalRef.componentInstance.bandId = roleBandId;
+    modalRef.componentInstance.capabilityId = roleCapabilityId;
+    modalRef.componentInstance.roleAdded.subscribe(data => {
+
+      if (data.data.hasOwnProperty('success')) {
+        this.showAlert = true;
```

Comment:
You can take this line out of the `if` and `else`, and just have `this.showAlert = true;` above the `if`.